File: calculadora.py
# Calculadora Versión 1.1

# Inicio: 20 de Diciembre 2025
# Fin: 

# Este programa realiza operaciones matemáticas básicas: suma, resta, multiplicación y división.
# Funciones actuales: Realiza las operaciones simples.
# Hecho por Karla

def sumar(a,b):
    """
    Función que se va a encargar de las sumas
    (float, float) --> float
    """
    
    return a+b

def restar(a,b):
    """
    Función que se va a encargar de las restas
    (float, float) --> float
    """
    
    return a-b

def multiplicar(a,b):
    """
    Función que se va a encargar de las multiplicaciones
    (float, float) --> float
    """
    
    return a*b

def dividir(a,b):
    """
    Función que se va a encargar de las divisiones
    Arroja error si el divisor es igual a 0 (b = 0)
    (float, float) --> float
    """

    return a/b 

def get_op(x):
    """
    Función que ayuda a identificar la operación
    (str) --> str
    """
    
    for i in x[1:]:
        if i in ["+","-","*","/"]:
            return i
    else:
        return "nini"

def strip_inter(x):
    """
    Función que actúa como .strip pero también remueve los espacios
    intermedios
    (str) --> str 
    """

    resultado = ""
    for i in x:
        if i != " ":
            resultado += i
    return resultado

def get_nums(x):
    """
    Función que ayuda a identificar los números
    Arroja error si no se encuentra operación
    (str) --> list
    """
    
    op = get_op(x)
    if op == "nini":
        return "No se ingresó ninguna operación"
    else:
        op_pos = x.index(op, 1)
        return [float(strip_inter(x[:op_pos])), float(strip_inter(x[op_pos+1:]))]


def main():
    cadena = input().strip()
    op = get_op(cadena)
    a = get_nums(cadena)[0]
    b = get_nums(cadena)[1]
    if op == "+":
        print(sumar(a,b))
    elif op == "-":
        print(restar(a,b))
    elif op == "*":
        print(multiplicar(a,b))
    elif op == "/":
        if b == 0:
            print("No es posible dividir por 0")
        else: 
            print(dividir(a,b))
    else:
        print("No se ingresó ninguna operación")
main()

File: domain/src/Calculadora.java
/*
* Calculadora

* Versión: 1.1
* Inicio: 21 de Diciembre del 2025
* Fin:

* Este programa realiza operaciones matemáticas básicas: suma, resta, multiplicación y división.
* Funciones actuales: Operaciones básicas + porcentaje.
* Hecho por: Karla
*/

public class Calculadora {

    /*
    * Do adds
    * @param a: a number to work with
    * @param b: another number to work with
    * 
    * @return: number equal to 'a' add 'b'
    */
    public float add(float a, float b){
        return a+b;
    }
    
    /*
    * Do subtraction
    * @param a: a number to work with
    * @param b: another number to work with
    * 
    * @return: number equal to 'a' subtract 'b'
    */
    public float minus(float a, float b){
        return a-b;
    }

    /*
    * Do multiplications
    * @param a: a number to work with
    * @param b: another number to work with
    * 
    * @return: number equal to 'a' times 'b'
    */
    public float times(float a, float b){
        return a*b;
    }

    /*
    * Do divisions
    * @param a: a number to work with
    * @param b: another number to work with
    * 
    * @return: the cocient between 'a' and 'b'
    * 
    * @throw: AritmeticException when divide by 0 (when 'b' = 0)
    */
    public float div(float a, float b){
        if (b == 0){
            throw new ArithmeticException("División por 0");
        }
        return a/b;
    }
    
    /*
    * Manage percentages
    * @param a: number for convertion
    * 
    * @return: number digited for 'a' divide by 100
    * 
    * @throw: AritmeticException when 'a' is negative
    */
    public float percentage(float a){
        if (a < 0){
            throw new ArithmeticException("% negativo");
        }
        return a / 100;
    }
}
